sips/sip-021/sip-021-trustless-two-way-peg-for-bitcoin.md

```diff
@@ -0,0 +1,1385 @@
+height monotonically increases.  When a new Stacks block is mined, its parent
+will either be the canonical Stacks tip, or it will be a sibling of an
+unconfirmed Stacks block.  In the first case, the highest frozen block-commit's
+height increases by 1.  In the second case, the highest frozen block-commit's
+height is unchanged.
+
+* When mining off of unblessed block-commits, the highest
+frozen block-commit at Bitcoin height N will either be the same as the one in
+Bitcoin height N+1, or it will be the parent of the one in Bitcoin height N+1.
+This is _not true_ when considering blessed block-commits.
+
+### Rules
+
+By introducing both lifecycle phases and Stacker blessings, this proposal
+creates a new fork-choice rule for Stacks: the canonical fork is always tipped
+by the highest descendant of the highest frozen block-commit's Stacks block,
+even if the block for it is not yet published.
+
+In doing so, the system creates a new miner rule: a miner can mine off of any
+descendant of the highest frozen block-commit (i.e. any unblessed
+block-commit), and it can mine off of any descendant of a blessed block-commit.
+Miners freely and individually choose which member of this set of block-commits
+to mine off of.  A miner cannot mine off of any other block-commits.
+
+### Finalization
+
+Once a block-commit receives 100 Bitcoin confirmations, it can no longer be
+blessed.
+
+When Stackers fulfill a peg-out request, they invalidate all Stacks forks in
+which the peg-out would result in an inconsistent chain state (i.e. someone's
+sBTC balance becomes negative).  These forks will never be treated as canonical
+as long as they contain such inconsistencies.  As such, the act of Stackers
+fulfilling a peg-out request that materializes in Stacks block B is declaring
+that any Stacks fork that descends from B might be canonical (i.e. the
+canonical fork must include B).
+
+Consequently, the act of processing a peg-out is used to keep the chain alive
+even if someone manages to mine 99 block-commits without disclosing their
+blocks.  As long as there exists at least one valid publicly-disclosed block,
+then Stackers can use it to authenticate and process peg-outs and in doing so,
+require that block to be part of the canonical chain.  This enables miners and
+Stackers to prevent a hidden-block miner from halting the chain while spending
+far fewer resources than the attacker.  It also prevents a 51% attacker from
+denying sBTC transfers and peg-outs: if any sBTC transfer materializes in the
+honest fork, and Stackers process at least one peg-out on that fork, then the
+51% attacker's fork is obliged to _also_ process the _same_ sBTC transfers in
+order to keep their fork canonical.
+
+# Related Work
+
+This section is not done yet.  Notes follow.
+
+* PowPeg in RSK
+
+	* Requires HSMs on a majority of Bitcoin miners (brittle;
+closed-membership)
+
+	* Has a token with coinbases to incentivize validator
+liveness regardless of tx volume (good!)
```

Comment:
PowHSM operators do not receive any sort of coinbase subsidy.

Reply:
They don't receive RIF?  News to me.

That makes their position weaker, then.  Why should validators care to stay online in periods of low activity, when there are other more-profitable things they could be doing at the same time?  That's the problem that a protocol-level coinbase solves (or some other minimum but fixed reward for just showing up).

Reply:
> They don't receive RIF?

Right. RIF is an application-layer asset. It's not used at all in the consensus layer.

> Why should validators care to stay online in periods of low activity, when there are other more-profitable things they could be doing at the same time?

The effort to merge-mine Rootstock is marginal: run a Rootstock node and set `generate` to `true` (exaggerating slightly but not far off from that level of simplicity). So the activity would have to be _really_ low for a _long_ period of time to make it worth it to stop merge-mining.